sdk/core/allocator/revoker.h

```diff
@@ -183,26 +183,40 @@ namespace Revocation
 			 * Otherwise, there are at least two words of the bitmap that need
 			 * to be updated with the masks, and possibly some in between that
 			 * just need to be set wholesale.
+			 *
+			 * We paint ranges "backwards", from highest address to lowest, so
+			 * that we never create a window in which an interior pointer has a
+			 * clear shadow bit while the lower adjacent address has an asserted
+			 * shadow bit, as that would open the door to confusing the interior
+			 * pointer with a pointer to the start of an object (recall that
+			 * object headers are marked in the shadow bitmap).
 			 */
 			WordT midWord;
 			if (fill)
 			{
-				shadowCap[baseWordIx] |= maskLo;
 				shadowCap[topWordIx] |= maskHi;
 				midWord = ~WordT(0);
 			}
 			else
 			{
-				shadowCap[baseWordIx] &= ~maskLo;
 				shadowCap[topWordIx] &= ~maskHi;
 				midWord = 0;
 			}
 
-			for (size_t shadowWordIx = baseWordIx + 1; shadowWordIx < topWordIx;
-			     shadowWordIx++)
+			for (size_t shadowWordIx = topWordIx - 1; baseWordIx < shadowWordIx;
```

Comment:
Could `topWordIdx` ever be `0`?

Reply:
I think we are safe from underflow because it could only happen if `topWordIdx == baseWordIdx` in which case we would already have returned in the case above. Maybe a coment to that effect would be good?